File: Lab3_c_sharp/domain/Cursa.cs
namespace Lab3_c_sharp.domain;

public class Cursa : Entity
{
    private string destinatie;
    private DateTime dataPlecare;
    private TimeSpan oraPlecare;
    private int numarLocuriDisponibile;
    private Autocar autocar;

    public Cursa(int id, string destinatie, DateTime dataPlecare, TimeSpan oraPlecare, int numarLocuriDisponibile, Autocar autocar) : base(id)
    {
        this.destinatie = destinatie;
        this.dataPlecare = dataPlecare;
        this.oraPlecare = oraPlecare;
        this.numarLocuriDisponibile = numarLocuriDisponibile;
        this.autocar = autocar;
    }
    
    public string Destinatie
    {
        get => destinatie;
        set => destinatie = value;
    }
    public DateTime DataPlecare
    {
        get => dataPlecare;
        set => dataPlecare = value;
    }
    public TimeSpan OraPlecare
    {
        get => oraPlecare;
        set => oraPlecare = value;
    }
    public int NumarLocuriDisponibile
    {
        get => numarLocuriDisponibile;
        set => numarLocuriDisponibile = value;
    }
    public Autocar Autocar
    {
        get => autocar;
        set => autocar = value;
    }
        

   
}

File: Lab3/src/main/java/org/example/domain/Cursa.java
package org.example.domain;

public class Cursa extends Entity<Long>{

    private String destinatie;
    private String dataPlecare;
    private String oraPlecare;
    private int nrLocuriDisponibile;
    private Autocar autocar;

    public Cursa(String destinatie, String dataPlecare, String oraPlecare, int nrLocuriDisponibile, Autocar autocar) {
        this.destinatie = destinatie;
        this.dataPlecare = dataPlecare;
        this.oraPlecare = oraPlecare;
        this.nrLocuriDisponibile = nrLocuriDisponibile;
        this.autocar = autocar;
    }

    public String getDestinatie() {
        return destinatie;
    }
    public void setDestinatie(String destinatie) {
        this.destinatie = destinatie;
    }
    public String getDataPlecare() {
        return dataPlecare;
    }
    public void setDataPlecare(String dataPlecare) {
        this.dataPlecare = dataPlecare;
    }
    public String getOraPlecare() {
        return oraPlecare;
    }
    public void setOraPlecare(String oraPlecare) {
        this.oraPlecare = oraPlecare;
    }
    public int getNrLocuriDisponibile() {
        return nrLocuriDisponibile;
    }
    public void setNrLocuriDisponibile(int nrLocuriDisponibile) {
        this.nrLocuriDisponibile = nrLocuriDisponibile;
    }
    public Autocar getAutocar() {
        return autocar;
    }
    public void setAutocar(Autocar autocar) {
        this.autocar = autocar;
    }

    @Override
    public String toString() {
        return "Cursa{" +
                "destinatie='" + destinatie + '\'' +
                ", dataPlecare='" + dataPlecare + '\'' +
                ", oraPlecare='" + oraPlecare + '\'' +
                ", nrLocuriDisponibile=" + nrLocuriDisponibile +
                ", autocar=" + autocar +
                '}';
    }
}
